environment/src/main/java/com/sequenceiq/environment/environment/validation/securitygroup/aws/AwsEnvironmentSecurityGroupValidator.java

```diff
@@ -76,19 +79,27 @@ private void checkSecurityGroupVpc(EnvironmentDto environmentDto, ValidationResu
         CloudSecurityGroups securityGroups = platformParameterService.getSecurityGroups(request);
 
         boolean securityGroupInVpc = false;
-        for (CloudSecurityGroup cloudSecurityGroup : securityGroups.getCloudSecurityGroupsResponses().get(region.getName())) {
-            Object vpcId = cloudSecurityGroup.getProperties().get("vpcId");
-            if (cloudSecurityGroup.getGroupId().equals(securityGroupId)) {
-                LOGGER.info("Security group {} was found on AWS side.", securityGroupId);
-                if (vpcId != null && vpcId.toString().equals(environmentDto.getNetwork().getAws().getVpcId())) {
-                    securityGroupInVpc = true;
-                    break;
+        String awsVpcId = environmentDto.getNetwork().getAws().getVpcId();
+        Set<CloudSecurityGroup> cloudSecurityGroups = securityGroups.getCloudSecurityGroupsResponses().get(region.getName());
+        for (String securityGroupId : getSecurityGroupIds(securityGroupIds)) {
+            securityGroupInVpc = false;
+            for (CloudSecurityGroup cloudSecurityGroup : cloudSecurityGroups) {
```

Comment:
This will yield an NPE if `cloudSecurityGroups` is `null`. I suggest wrapping the `for` statement inside a `if Objects.nonNull(cloudSecurityGroups))` condiiton (like for `AzureEnvironmentSecurityGroupValidator` and `GcpEnvironmentSecurityGroupValidator`).